js/scenes/UserProfileCreate/UserProfileCreate.js

```diff
@@ -19,6 +25,39 @@ const UserProfileCreate = ({ userID }) => (
   <ScrollView>
     <View>
```

Comment:
You're going to need a heftier form solution here. Calling Firebase methods with each change in the text input is not ideal (...performance implications, plus each change counts as one request to the API, which could get expensive with a lot of users).

Even if autosave is a critical feature, you would want to debounce these requests, or perhaps only make them on field blur. So here are your options:

*Option 1*

Use Redux for managing your form input state and use Redux Form, if you most comfortable with that after Boomtown.

*Option 2*

Avoid Redux and use React Final Form, which will manage your form state for you:

https://github.com/final-form/react-final-form

You can then use the `handleSubmit` to run your Firebase methods, which will be collected in one `Promise.all` in your new helper function (see above).